AEPMessaging/Sources/Messaging.swift

```diff
@@ -117,17 +117,31 @@ public class Messaging: NSObject, Extension {
     /// Generates and dispatches an event prompting the Edge extension to fetch in-app messages.
     /// The app surface used in the request is generated using the `bundleIdentifier` for the app.
     /// If the `bundleIdentifier` is unavailable, calling this method will do nothing.
-    private func fetchMessages() {
+    private func fetchMessages(for surfaces: [String]? = nil) {
         guard appSurface != "unknown" else {
-            Log.warning(label: MessagingConstants.LOG_TAG, "Unable to retrieve in-app messages - unable to retrieve bundle identifier.")
+            Log.warning(label: MessagingConstants.LOG_TAG, "Unable to retrieve in-app or feed messages, cannot read the bundle identifier.")
+            return
+        }
+        
+        var targetSurfaces: [String] = []
+        if let surfaces = surfaces {
+            targetSurfaces = surfaces
+                .filter { !$0.isEmpty }
+                .map { appSurface + MessagingConstants.PATH_SEPARATOR + $0 }
+        } else {
+            targetSurfaces = [ appSurface ]
+        }
+        
+        if (targetSurfaces.isEmpty) {
+            Log.debug(label: MessagingConstants.LOG_TAG, "Unable to retrieve feed messages, no valid surface paths found.")
             return
```

Comment:
since this check really only applies to the flow where `surfaces` are provided as a param in this API, i'd prefer to see it up in the block that starts on line 127